### ISSUE TITLE
Failed to read torrent.

### ISSUE THREAD

**Issue by Jaseku** [Mika-/torrent-control#98]
I got this info in private torrent.
Can i do someting?

![fail](https://user-images.githubusercontent.com/11813376/86179635-eb5d4400-bb2a-11ea-9856-a99823b71a17.jpg)

**Comment by ryanerwin** [comment]
Probably need to click through to another page... That error usually comes when you try to load a regular page as a torrent...

Most likely the link you have is serving up another page not a .torrent or magnet link...

**Comment by Mika-** [comment]
Yes, you are trying to add a web page. This often happens when you accidentally click a link to a torrents details page instead of torrent.

**Comment by Antorell** [comment]
It also happens when the link to the .torrent file is behind a DDOS protection (cloudflare, etc..).

**Comment by keylogthis** [comment]
> It also happens when the link to the .torrent file is behind a DDOS protection (cloudflare, etc..).

Yeah, this started happening for me on a site that uses cloudflare, a few weeks ago it was working fine.

**Comment by StarvingDeveloper** [comment]
I have same problem with private tracker  - when I disable addon I am getting torrent file donwload dialog and when enable addon I am getting that error message.
And since other addons dont have such problem, maybe you could review some permissions or something - your addon way better as competitors, but fail main task to read torrent file from such links:
<censored>.net/download.php?id=903733&name=Griftlands.v484799-PLAZA.torrent

**Comment by Mika-** [comment]
Any current examples of failing urls? Preferrably on public sites.
I'm experimenting a new torrent fetching implementation and would like to test it with some problematic scenarios.

**Comment by Akatokage** [comment]
Anything from https://rarbgaccess.org for example. Magnet links work, but not torrent links.

**Comment by Akatokage** [comment]
Anything from https://rarbgaccess.org/ for example. Magnet links work, but not torrent links. I don't know if its just a mime type problem because it gets generated by php as html or something. Maybe it's just qBittorrent that's the problem for me.

**Comment by Mika-** [comment]
Can't find anything wrong with that site. Everything adds to qbittorrent fine. Mime types and other headers are ok and cloudflare allows traffic to pass even without cookies.
Site does do some heavy clickjacking and redirecting. So it could be that some ads have urls that are made to look like torrents to fool people to click them.

**Comment by Akatokage** [comment]
Doesn't work for me, I have version 0.2.29 of the addon. Without the addon the torrent file downloads fine on a left click, but when I right click and save as, a threat_defence.htm gets downloaded. I guess that explains the type problem for me. I even tried it on a fresh Firefox install Version 105.0.3 (64-bit) with only the addon installed, same problem.

**Comment by Mika-** [comment]
That seems to be some custom anti scraping system they are using. It could be triggered by ip or other network information as i'm not encountering it.

I made [experimental build](https://github.com/Mika-/torrent-control/raw/contentscript/torrent_control-dev-0.2.29.xpi) of the addon if you want to check if it behaves differently.

**Comment by Akatokage** [comment]
With the experimental build there isn't even a notification. Magnet links and other sites work fine. But your guess about IP's was probably right, when I change my IP to one in the US it started working even with the default addon.

Sorry for the trouble and thank you.

**Comment by StarvingDeveloper** [comment]
> That seems to be some custom anti scraping system they are using. It could be triggered by ip or other network information as i'm not encountering it.
> 
> I made [experimental build](https://github.com/Mika-/torrent-control/raw/contentscript/torrent_control-dev-0.2.29.xpi) of the addon if you want to check if it behaves differently.

Unfortunately all trackers who failed previously is private, so I just installed this experimental build and it worked!

**Comment by Mika-** [comment]
After some additional testing reworked torrent fetching is now available in the latest version.

**Comment by Purgatory8593** [comment]
As of today, the latest version 0.2.30 seems to have partially broken torrent fetching for me. It still worked on a site where the link directly points to a .torrent file, but otherwise gives this error on a private site which has worked just fine up until now:

`Failed to read torrent. (Unknown type: application/x-bittorrent; charset=utf-8)`

I am on Arch Linux, using the latest version of Firefox, and I am connected to a remote qBittorrent instance using HTTP authentication. I have had no problems with the site in question up until this release.

The links that now fail, end with something like this:

`/torrents.php?action=download&id=xxx&authkey=xxx&torrent_pass=xxx`

**Comment by Mika-** [comment]
Released a updated version with loosened type check that fixes that. Torrent files are always utf-8 so it's not common to include that information.

**Comment by keylogthis** [comment]
I just wanted to update that the site I had previously had a problem with is now working with this update, thank you.

**Comment by victoruptons** [comment]
I have the same problem in one of the private tracker site. I try to manually paste the URL directly to the qBittorrent's "add torrent link" menu and the link does work. But if I right click and choose the extension's add torrent, it will show the unknown type: text/html error.

**Comment by StarvingDeveloper** [comment]
I can acknowledge it stopped work on some trackers. But on different trackers as before - after last fix they still work. 
This time for example rutor.org spit same text/html error. Looks like plugin need some update again.
